Source/WebCore/page/Location.cpp

```diff
@@ -255,6 +255,11 @@ ExceptionOr<void> Location::replace(LocalDOMWindow& activeWindow, LocalDOMWindow
     if (!activeWindow.document()->canNavigate(dynamicDowncast<LocalFrame>(frame.get()), completedURL))
         return Exception { SecurityError };
 
+    // If the loader for activeWindow's frame (browsing context) has no outgoing referrer, then set the loader's
+    // outgoing referrer to the URL of firstWindow's frame's (browsing context's) Document — which is the
+    // incumbent global object's associated Document. https://html.spec.whatwg.org/#location-object-navigate
+    if (activeWindow.frame()->loader().outgoingReferrer().isEmpty())
+        activeWindow.frame()->loader().setOutgoingReferrer(firstFrame->document()->urlForBindings());
```

Comment:
It seems this doesn't take referrer policy into account. Wouldn't that result in problems?

Reply:
> It seems this doesn't take referrer policy into account. Wouldn't that result in problems?

Not sure; I’ll take a look at the other existing code that does use referrer policy, and see if I can update this to follow what the code is doing in other places for referrer policy

Reply:
> Hmm, I haven't looked, but maybe that happens later and this is fine therefore. Not sure.

Yeah, after some initial looking, I’m pretty sure it does happen later — but I’m stepping through the code now and will make certain that it actually happens later, and then post a comment with the details.

Reply:
As far as referrer-policy handling, it seems things so something like this:
```
Location::replace
  LocalDOMWindow::setLocation
    NavigationScheduler::scheduleLocationChange
      FrameLoader::changeLocation
        FrameLoader::loadFrameRequest
```
…and in `FrameLoader::loadFrameRequest`, here:

https://github.com/WebKit/WebKit/blob/8a62a65095faf477ebaeb47b9d236b92a4d910d1/Source/WebCore/loader/FrameLoader.cpp#L1317-L1320

…is where any referrer policy gets evaluated, and the `Referer` header is actually set (or not) based on the policy.